### DISCUSSION TITLE
I found it could not support other language like chinese character, could it be possible to print the chinese character output? Seems only support english character output.

### DISCUSSION THREAD

**Discussion by dustreturn** [pharmaverse/formatters#220] [Q&A]
/

**Comment by shajoezhu** [comment]
hi @dustreturn , perhaps with encoding issue. I was wondering if you could provide us an example and we can investigate further. Thanks!

**Reply by dustreturn** [reply]
Hi Joe, thanks for reply, I've kind of solved this probelm by using the package flextable, but still let me share the code as below.



`lyt <- basic_table(title="Table 14.3.3.1.1 Summary of Laboratory Results and Change from Baseline by Visit (<Lab Category>)  - Safety Set",show_colcounts = TRUE) %>%
  split_cols_by("ACTARM",ref_group = '安慰剂') %>%
  split_rows_by("PARAM", split_fun = split_fun, label_pos = "topleft", split_label = obj_label(adlb1$PARAM)) %>%
  split_rows_by("AVISIT", split_fun = split_fun, label_pos = "topleft", split_label = obj_label(adlb1$AVISIT)) %>%
  split_cols_by_multivar(
    vars = c("AVAL", "CHG",'pvalue1'),
    varlabels = c("观察值", "与基线改变值",'组内与基线比较P值')
  ) %>%
  analyze_colvars(afun = afun)

result <- build_table(lyt, adlb1, alt_counts_df = randlist3)
result_pgby<-paginate_table(result,lpp=40)



afun <- function(x, .var, .spl_context, ...) {
  n_fun <- sum(!is.na(x), na.rm = TRUE)
  
  is_pvalue1 <- .var == 'pvalue1'
  is_pvalue2 <- .var == 'pvalue2'
  is_pbo <- .spl_context$value[which(.spl_context$split == "ACTARM")] == "安慰剂"
  
  if (n_fun == 0) {
    mean_fun1 <- NA
    mean_fun2 <- NA
    my_fun <- NA
    mean_sd_fun <- c(NA, NA)
    median_fun <- NA
    min_max_fun <- c(NA, NA)
  } else {
    mean_fun1 <- mean(x, na.rm = TRUE)
    mean_fun2 <- mean(x, na.rm = TRUE)
    my_fun <- function(x) {
      if (is_pbo) {
        return(NA)
      } else {
        return(mean(x))
      }
    }
    mean_sd_fun <- c(mean(x, na.rm = TRUE), sd(x, na.rm = TRUE))
    median_fun <- median(x, na.rm = TRUE)
    min_max_fun <- c(min(x), max(x))
  }
  is_chg <- .var == "CHG"
  is_baseline <- .spl_context$value[which(.spl_context$split == "AVISIT")] == "Baseline"
  
  if (is_baseline && is_chg) n_fun <- mean_sd_fun <- median_fun <- min_max_fun <- NULL
  if (is_pvalue1) n_fun <- mean_sd_fun <- min_max_fun <- NULL
  if (is_pvalue2) n_fun <- mean_sd_fun <- min_max_fun <- NULL
  
  if (is_baseline) mean_fun1 <- NULL
  if (is_baseline) mean_fun2 <- NULL
  is_aval <- .var == 'AVAL'
  if (is_chg) mean_fun1 <- NULL
  if (is_aval) mean_fun1 <- NULL
  if (is_chg) mean_fun2 <- NULL
  if (is_aval) mean_fun2 <- NULL
  
  #if (is_pvalue2 & is_pbo) mean_fun2 <- NULL
  
  in_rows(
    "n" = n_fun,
    "Mean (SD)" = mean_sd_fun,
    "Median" = median_fun,
    "Min - Max" = min_max_fun,
    #"P value" = mean_fun1,
    #"P value 2" = mean_fun2,
    .formats = list("n" = "xx", "Mean (SD)" = "xx.xx (xx.xx)", "Median" = "xx.xx", "Min - Max" = "xx.xx - xx.xx"
                    #,"P value" = "xx.xxxx","P value 2" = "xx.xxxx"
    ),
    .format_na_strs = list("n" = "NE", "Mean (SD)" = "NE (NE)", "Median" = "NE", "Min - Max" = "NE - NE")
  )
}

lyt2 <- basic_table(title="Table 14.3.3.1.1 Summary of Laboratory Results and Change from Baseline by Visit (<Lab Category>)  - Safety Set",show_colcounts = TRUE) %>%
  #split_cols_by("ACTARM",ref_group = '安慰剂') %>%
  split_rows_by("PARAM", split_fun = split_fun, label_pos = "topleft", split_label = obj_label(adlb1$PARAM)) %>%
  split_rows_by("AVISIT", split_fun = split_fun, label_pos = "topleft", split_label = obj_label(adlb1$AVISIT)) %>%
  split_cols_by_multivar(
    vars = c('pvalue2'),
    varlabels = c('组间比较P值')
  ) %>%
  analyze_colvars(afun = afun)
adlb3 <- adlb1 %>% mutate(ACTARM='')
result2 <- build_table(lyt2, adlb3, alt_counts_df = randlist3,col_widths = c(100,100))
result3 <- cbind_rtables(result,result2)
#result3 <- paginate_table(result3,lpp=40)
#options(encoding ="UTF-8")
#a <- matrix_form(result)

export_as_rtf(result3,landscape = TRUE,max_width = 40, file="E:/Ranalysis/Routput/aaa.rtf",tf_wrap = TRUE,colwidths = c(20,20,20,20,20,20,20,20),font_family="Simsun")
export_as_txt(result3,landscape = TRUE,max_width = 40, file="E:/Ranalysis/Routput/aaa.txt",tf_wrap = TRUE,colwidths = c(15,15,15,15,15,15,15,15))
`
export_as_rtf is not work, the chinese character of label and information from dataset are all corrupted characters.
export_as_txt has no probelm.
export_as_pdf has the probelm as export_as_rtf.
